### ISSUE TITLE
Support for Unicode characters

### ISSUE THREAD

**Issue by boris-noyvert** [TeWu/GmailAttachmentsExtractor#10]
I wonder if email subjects containing Unicode characters (from non-latin languages) can be preserved in the corresponding name of the folder for the attachments? I think Unicode characters are substituted by "_" in the folder name at the moment. 

Just to be clear - the attachment files containing Unicode characters in the name are saved fine, with the same name as in the email.

Otherwise I am really happy with v1.0.2 - works very well, I didn't encounter any problems so far. 
Thank you for your efforts!

**Comment by TeWu** [comment]
Thanks for reporting.

When prototyping GmailAttachmentsExtractor I went the easy way there, and made the program only preserve Polish characters, that I cared about 😉. That solution survived to the first, and few subsequent releases... The good news is, that I finally upgraded the file system objects name sanitization algorithm 😄.
The new algorithm should preserve all 132875 [Unicode alphabetic characters](https://www.unicode.org/reports/tr44/#Alphabetic). The algorithm kicks in only when the file system has a problem with the path - so the program should even preserve directory and file names with non-alphabetic characters, as long as file system is happy with it.
Sorry for the inconvenience, and I hope those improvements will make the program even more useful 😃.

---

I created GmailAttachmentsExtractor, thinking that it would be a program that one can use once to free some space from nearly-full Gmail inbox, and then don't use it for some time - until the next time the inbox is nearly-full. But it seems, like You are using it differently - which makes me curious, how do You use GmailAttachmentsExtractor?
If you can drop me a short email about it, that would be great 😃. 
My email: `tewu.dev@gmail.com`.

**Comment by TeWu** [comment]
I've compiled the new version for your convenience.
**[GmailAttachmentsExtractor_v1.0.2.next_2021.04.04.zip](https://github.com/TeWu/GmailAttachmentsExtractor/files/6254943/GmailAttachmentsExtractor_v1.0.2.next_2021.04.04.zip)**

**Comment by boris-noyvert** [comment]
Thank you, Unicode characters work fine now. However now there is another problem caused by "/" and "." characters in the email subject (running from Windows 10 command prompt).

"/" in the subject cuts off all the preceding characters (including the date and time) from the folder name. For example, from an email with subject "Journal Club (10/12/15)" the following folder is created: "15)", or subject "Parent/Carer Forum - March 2018" ends up in folder "Carer Forum - March 2018". I don't think this was the case before the Unicode fix.

Also I think multiple "." in the subject cause the following error:

```
1076/~2021 (53%) | Processing email 'one more analysis.........'
    Extracting 1 attachment(s) to directory '2014.09.23 16_17_04 one more analysis.........'
java.io.FileNotFoundException: C:\Users\Boris\Downloads\GmailAttachmentsExtractor_v1.0.2.next_2021.04.04\gmail_cleanup_2021-04-04\2014.09.23 16_17_04 one more analysis.........\Custom_Track_predictions.txt (The system cannot find the path specified)
        at java.io.FileOutputStream.open0(Native Method)
        at java.io.FileOutputStream.open(Unknown Source)
        at java.io.FileOutputStream.<init>(Unknown Source)
        at pl.geek.tewu.gmail_attachments_extractor.Utils.openOutputStream(Utils.java:168)
        at pl.geek.tewu.gmail_attachments_extractor.Utils.copyInputStreamToFile(Utils.java:147)
        at pl.geek.tewu.gmail_attachments_extractor.GmailAttachmentsExtractor.saveToFile(GmailAttachmentsExtractor.java:299)
        at pl.geek.tewu.gmail_attachments_extractor.GmailAttachmentsExtractor.extractAttachments(GmailAttachmentsExtractor.java:167)
        at pl.geek.tewu.gmail_attachments_extractor.Main.call(Main.java:45)
        at pl.geek.tewu.gmail_attachments_extractor.Main.call(Main.java:14)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1783)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2150)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2144)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
        at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:1975)
        at picocli.CommandLine.execute(CommandLine.java:1904)
        at pl.geek.tewu.gmail_attachments_extractor.Main.main(Main.java:29)
```

I don't know if this problem was present before the Unicode fix.

**Comment by TeWu** [comment]
Thanks, for reporting again :)
OK, so not everything working smooth yet; I'm working on it..

**Comment by TeWu** [comment]
OK, fixed in 43236f6.
Sanitization for file system objects with names containing `/` and `.` should work fine now.

Compiled version with those changes:
**[GmailAttachmentsExtractor_v1.0.2.next_2021.04.07.zip](https://github.com/TeWu/GmailAttachmentsExtractor/files/6273892/GmailAttachmentsExtractor_v1.0.2.next_2021.04.07.zip)**
